src/manifest.c

```diff
@@ -1427,3 +1427,127 @@ int verify_image(struct image *image)
 	fclose(in_file);
 	return 0;
 }
+
+
+int resign_image(struct image *image)
+{
+	int key_size, key_file_size;
+	size_t size, read;
+	FILE *in_file;
+	void *buffer;
+	int ret, i;
```

Comment:
rimage does not live in the previous century like the kernel does, so you can move this at first point of use and combine declaration and initialization as allowed by C99

Reply:
I guess all this goes away if just setting this to NULL at declaration as clang suggest... I mean it is a clear bug to free some bogus

Reply:
@marc-hb made this: https://github.com/thesofproject/rimage/pull/78.

Reply:
Initializing all variables with some default value at the top makes sense... when and only when some default value actually makes sense. In other cases it merely hides compiler warnings finding bugs for C99 backward-compatibility (I didn't read that code in that PR).